pkg/ui/static/js/bucket.js

```diff
@@ -71,6 +77,12 @@ function draw() {
     }
 }
 
+// isThanosBlock eturns true if it is a block that has been processed by Thanos.
+function isThanosBlock(d) {
+    return d.thanos.labels !== null &&
```

Comment:
Is it guaranteed that non-thanos blocks have a `thanos` field at all? In other words, can we be certain that `d.thanos !== undefined`? Or do we need to do `d.thanos && d.thanos.labels !== null && d.thanos.source !== ""`?

Reply:
Yep: https://github.com/thanos-io/thanos/blob/master/pkg/block/metadata/meta.go#L59

Since it is not a reference, it means that `d.thanos` will always exist. We might want to change this but then this change will be visible to the end user. I think we should convert it into a reference in the future PR where I (we) will make the bucket viewer UI work with non-Thanos blocks.

Zero value for `labels` is `nil` and for `source` it is an empty string so we don't need to do this. All of the other struct members have `omitempty` so they do not appear at all. It should be enough to check these.

Reply:
We can use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) here to handle the case when `d.thanos` is `undefined`.
```suggestion
    return d.thanos?.labels !== null &&
```